src/en/charms-constraints.md

```diff
@@ -1,187 +1,262 @@
-constraints (`juju add-machine`) and then specify that machine when deploying 
-applications ([see the documentation on `juju deploy`](./charms-deploying.html)).
+Several noteworthy constraint characteristics:
 
-## What constraints can be used?
+ - Some constraints are only supported by certain clouds.
+ - Changes to constraint defaults do not affect existing machines.
+ - Multiple constraints are logically AND'd (i.e. the machine must satisfy all
+   constraints).
 
-There is a full list of the constraints used 
-[in the reference section](reference-constraints.html). Be aware that some of 
-these are specific to the type of cloud you are using. For example, one may
-understand an "instance-type" constraint, but another may not. 
+## Clouds and constraints
 
-The most useful constraints for Juju in general are:
-  
-  - **mem** : This indicates the minimum number of megabytes of RAM that must 
-  be available to an application unit. An optional suffix of M/G/T/P indicates
-  the value is mega-/giga-/tera-/peta- bytes.
-
-  - **cores** :  How many cpu cores the host machine should have. This is a
-  crude indicator of system performance.
-    
-  - **spaces** : Target a particular network space, or avoid one. See 
-  [Network spaces][network-spaces].
-  
-  - **arch** : Short for 'architecture', indicates the processor type an
-  application must run on. One of amd64, arm, i386, arm64, or ppc64el.
-  
-With these you can make sure an application has the resources it needs to run 
-properly.
+The idealized use case is that of stipulating a constraint when deploying an
+application and the backing cloud providing a machine with those exact
+resources. In the majority of cases, however, default constraints may have been
+set (at various levels) and the cloud is unable to supply those exact
+resources.
 
-Constraints can be used with commands that support the '--constraints' option. 
-These are covered in more detail below, but there are some aspects of specifying
-constraints that are worth mentioning first.
+When the backing cloud is unable to precisely satisfy a constraint, the
+resulting system's resources will exceed the constraint-defined minimum.
+However, if the cloud cannot satisfy a constraint at all then an error will be
+emitted and a machine will not be provisioned.
 
-In the following examples, we will be using the `juju deploy` command, as this
-is the simplest and most frequent case for using constraints. So, to deploy the
-'mariadb' application to a machine with 4 gigabytes of memory or more:
-  
-```bash
-juju deploy mariadb --constraints mem=4G
-```
+When using the localhost cloud, constraints are ineffectual due the nature of
+this cloud's underlying technology (LXD), where each machine will, by default,
+have access to **all** of the LXD host's resources. Here, an exact hardware
+specification can be requested, but is done at the LXD level. An example will
```

Comment:
...at the LXD level (see example below).

I think is better

Reply:
:ok: